### PR TITLE
bugfix: delete should not show invalid rest responses message

### PR DESCRIPTION
**Pull request by Simperfit**

### CODE REVIEW

File: src/hydra/hydraClient.js [hunk 1]
```diff
@@ -302,7 +302,7 @@ export default ({entrypoint, resources = []}, httpClient = fetchHydra) => {
           .then(data => ({data, total: response.json['hydra:totalItems']}));
 
       case DELETE:
-        return Promise.resolve(() => ({data: {}}));
+        return Promise.resolve(response => (response.data: {}));
```
Comment (soyuka) [inline src/hydra/hydraClient.js]:
```
Promise.resolve(response => response.data)
```

File: src/hydra/hydraClient.js [hunk 2]
```diff
@@ -302,7 +302,7 @@ export default ({entrypoint, resources = []}, httpClient = fetchHydra) => {
           .then(data => ({data, total: response.json['hydra:totalItems']}));
 
       case DELETE:
-        return Promise.resolve(() => ({data: {}}));
+        return Promise.resolve(response => (response.data));
```
Comment (soyuka) [inline src/hydra/hydraClient.js]: parenthesis not needed in the `(response.data)`

Comment (mauchede) [reply src/hydra/hydraClient.js]: Or you can use `destructuring`:

```js
({ data }) => data
```

Comment (k911) [reply src/hydra/hydraClient.js]: Just to notice: in DELETE_MANY you just do basically:

```return {data: {}};```

which fixes also this error :smile:

Comment (Simperfit) [reply src/hydra/hydraClient.js]: You can delete without using delete_many

Comment (Simperfit) [reply src/hydra/hydraClient.js]: @mauchede using destructuring generate more codes with yarn build than just this.

Comment (soyuka) [reply src/hydra/hydraClient.js]: @Simperfit is no js-hipster :cry:

**Comment by k911** [comment]
Is there any chance that this fix will be finished and released soon? On admin application with enabled authorization, this error somehow causes further request to not have an authorization header which leads to user logout. When fix is applied everything works correctly.

**Comment by Simperfit** [comment]
yarn fix has been applied, waiting for CI then merging this.

**Comment by k911** [comment]
@Simperfit It seems it's still formatted unproperly.

**Comment by k911** [comment]
Hi again @Simperfit @dunglas @mauchede, any news on this? I can submit PR that passes CI checks, if you want.

This bugfix is important to me, because apparently after deleting single element, admin won't send 'Authorization' headers in next requests and it makes user logged out, because of failed request.

**Comment by dunglas** [comment]
@k911 it would be nice!

**Comment by k911** [comment]
This can be closed, I think.

**Comment by Simperfit** [comment]
Sorry I was on vacation ;)

**Pull request closed without merging**